addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -312,6 +312,8 @@ def _search(self, domain, offset=0, limit=None, order=None, access_rights_uid=No
         """
         employee = self.env['hr.employee']._get_contextual_employee()
         if order == self._order and employee:
+            if 'default_date_from' in self._context:
```

Comment:
We can't do the change in the context in `onchange` because the issue is not happening in this call it happens when we click on the dropdown menu in the popup and get all the leave types so those are two different requests and the context will be cleaned after getting the popup

Also for the `self = self.with_context()` I tried it and it won't be kept for the whole request since the `self` instance will be changing I assume during the request